### DISCUSSION TITLE
Long objects disappear when navigating in the scene

### DISCUSSION THREAD

**Discussion by 640kb** [pelicanmapping/osgearth#2183] [Support Q&A]
Hello,

we have a problem with displaying object data.
The objects are loaded into the scene with a FeatureModel layer as ModelSymbol.

The problem is that when navigating in the scene, sometimes whole objects are not rendered anymore.
This happens especially with objects that are very long. We have an example where an object is four times as long as the tile size we defined in the layout. We suspect that by culling process it is decided that these will not be rendered anymore.

Attached are two screenshots that show the problem again. On the first picture you can see a red line. If you move the scene slightly to the right, this object will no longer be rendered as you can see in the second image.

![osgearth-3](https://user-images.githubusercontent.com/708255/203553108-96cc0b6b-4b53-4c91-9098-d895389e554c.PNG)

![osgearth-4](https://user-images.githubusercontent.com/708255/203553145-40c6ca0c-fd89-4cf7-9092-c4bdcbc053ea.PNG)

If we understand it correctly, data from FeatureModels is managed in the PagingManager. In the overridden traverse function it is checked if the PageNodes are still within a certain range. But unfortunately the behavior could not be traced here, that when moving the scene certain objects are not rendered anymore.

This problem occurs if we define a layout with a tile_size in the earth file. And unfortunately we can't split the long objects into smaller objects.

Attached is a small minimal example that can be used to reproduce the problem. If you move the scene along the red line from the first viewpoint, then the object disappears from a certain point. Alternatively, you can jump to viewpoint 2 and move the scene a bit to the left so that the line becomes visible again.

Can anyone explain what triggers this behavior? Or has an idea for a solution to the problem?

[example.zip](https://github.com/gwaldron/osgearth/files/10075740/example.zip)

**Comment by gwaldron** [comment]
Sorry for the delayed response. Got a chance to test this, and the only issue was the object disappearing on zoom-out. This is remedied with a `max_range="1e10"` on the FeatureModel layer.
